packages/driver/src/cypress/command_queue.ts

```diff
@@ -557,4 +559,27 @@ export class CommandQueue extends Queue<$Command> {
 
     return promise
   }
+
+  // This function iterates through all upcoming commands in the queue, then
+  // discards the subject chain for every chainer that can't be referenced
+  // in the future (eg, no upcoming commands belong to the same chain).
+
+  // This is safe because aliases (which might be referenced later) are stored
+  // separately, in state('aliases'), and any subjects that "flow upwards" (eg.
+  // the subject of a chain inside a .then() command) have already replaced
+  // the subject of their parent chainer by the time this is called.
+  cleanSubjects () {
+    const stillNeeded = this.queueables.slice(this.index).map((c) => c.get('chainerId'))
+
+    this.queueables.slice(0, this.index).forEach((command) => {
+      // Once a command has resolved, and its chainer is no longer referenced
+      // by future commands, we can throw away the reference to the function
+      // and its subject to free memory.
+      if (command.get('subject') && stillNeeded.indexOf(command.get('chainerId')) === -1) {
+        command.set({ fn: null, subject: null, queryFn: null })
```

Comment:
why does the command store a reference to the subject at all? Is this being used?

Reply:
Are we able to drop userInvocationStack as well? At this point we don't necessarily need the stack details since the command has finished.

Reply:
The subject is used in three places:
- `src/cypress/log.ts` - The default `consoleProps` function reads the subject from command.get('subject') to display it as 'Yielded'. I think this could probably be redone. How often is the default consoleProps not overridden? And when it isn't, is there a better place to get this? Not sure.
- - `src/cy/logGroup.ts` - in forwardYieldedSubject. I think this could probably be rewritten / removed, but haven't spent time looking at logGroups to know how they working.
- `src/cypress/command_queue.ts` - The command queue itself. In `commandRunningFailed`, we look at `current.get('prev').get('subject')` to set the consoleProps. This could be read from cy.state('subjects') rather than the command's individual subject.
- One of our tests, `cypress/e2e/commands/assertions.cy.js`. Could be rewritten easily, 100% sure it's not needed here.

So quick census says "we probably don't need to anymore, but I'm uncomfortable trying to remove it just before release, would need more investigation."

Reply:
I was looking at `userInvocationStack` as well, but got scared off by the number of places it's used and my unfamiliarity with them. It doesn't look bad, but our stack traces are pretty fragile and not well tested. Any changes there I'd want to spend time digging through to understand more carefully.

The subject, fn and queryFn I understand fairly well, so feel confident messing with on short notice.

Reply:
@BlueWinds Can we log issues to clean these up? I would have though these would have all been removed with the detached dom work & would be nice to remove these stale reference if we can! heavy bloat for large doms.

Reply:
Logged https://github.com/cypress-io/cypress/issues/24903 to cover that, good idea. Also created https://github.com/cypress-io/cypress/issues/24904 to cover some other work in the area I'd like to do.